### PR TITLE
Byte add binary operator

### PR DESCRIPTION
**Pull request by kartikadur**
Fixed the `__str__` function in Bytes.js from comparing str with ascii (int val) to comparing two ascii/unicode values. Added code to check addition for Bytes with Bool, Bytes, Dict, Float, Int, List, Set, Str, Tuple.

### CODE REVIEW

File: tests/datatypes/test_bytes.py [hunk 1]
```diff
@@ -105,23 +105,23 @@ class BinaryBytesOperationTests(BinaryOperationTestCase, TranspileTestCase):
     data_type = 'bytes'
 
     not_implemented = [
-        'test_add_bool',
+        # 'test_add_bool',
```
Comment (swenson) [inline tests/datatypes/test_bytes.py]: I would just delete these lines rather than comment them out. We won't need to add them back in again hopefully :)

Comment (kartikadur) [inline tests/datatypes/test_bytes.py]: I making the changes now, so I had a question: is there a way to resubmit a pull request after commiting changes or do I need to create a new pull request everytime?

**Comment by kartikadur** [comment]
changes made and resubmitted. let me know if there are any other changes required.

**Comment by kartikadur** [comment]
awesome! thank you

**Pull request merged**